state/runtime/runtime.go

```diff
@@ -67,11 +67,37 @@ type Host interface {
 	GetTxContext() TxContext
 	GetBlockHash(number int64) types.Hash
 	EmitLog(addr types.Address, topics []types.Hash, data []byte)
-	Callx(*Contract, Host) ([]byte, uint64, error)
+	Callx(*Contract, Host) *ExecutionResult
 	Empty(addr types.Address) bool
 	GetNonce(addr types.Address) uint64
 }
 
+// ExecutionResult includes all output after executing given evm
+// message no matter the execution itself is successful or not.
+type ExecutionResult struct {
+	ReturnValue []byte // Returned data from the runtime (function result or data supplied with revert opcode)
+	GasLeft     uint64 // Total gas left as result of execution
+	GasUsed     uint64 // Total gas used as result of execution
+	Err         error  // Any error encountered during the execution, listed below
+}
+
+func (r *ExecutionResult) Succeeded() bool { return r.Err == nil }
+func (r *ExecutionResult) Failed() bool    { return r.Err != nil }
+func (r *ExecutionResult) Reverted() bool  { return r.Err == ErrExecutionReverted }
+
+func (r *ExecutionResult) CalculateGasUsed(gasLimit uint64, refund uint64) {
```

Comment:
It may be more proper name begins with `Update`or`Apply`?

Reply:
👍 https://github.com/0xPolygon/polygon-sdk/pull/103/commits/6d20c6500cdc85274366e5029673535f37cefca3